### PR TITLE
bndlib: Correct isValid check for workspace to look for build.bnd in the...

### LINKED ISSUE #874: Workspace.isValid check is incorrect
isValid looks for build.bnd to be at the root of the workspace instead of inside the build directory (cnf).

### PR DESCRIPTION
**Pull request by dhumeniuk**
... cnf folder.

Also, update rest of class to use constant instead of build.bnd string literal

Fixes #874

Signed-off-by: Humeniuk, Dave <david.humeniuk@udri.udayton.edu>

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java [hunk 1]
```diff
@@ -758,7 +758,7 @@ public File getBuildDir() {
 	}
 
 	public boolean isValid() {
-		return getFile(BUILDFILE).isFile();
+		return IO.getFile(buildDir, BUILDFILE).isFile();
```
Comment (dhumeniuk) [inline biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java]: getFile used the base workspace diretory

**Comment by dhumeniuk** [comment]
If there's a good place to create unit/integration tests, point me in the direction and I will try that. Just wasn't sure how all this work when the code accesses files directly. As in, is there a framework to mock out filesystem stuff for unit tests?

**Comment by bjhargrave** [comment]
> If there's a good place to create unit/integration tests, point me in the direction

Did you look at https://github.com/bndtools/bnd/blob/master/biz.aQute.bndlib.tests/src/test/WorkspaceTest.java? Most of the unit tests for bndlib are in the separate bndlib.tests project.

**Comment by dhumeniuk** [comment]
I'm unable to build the biz.aQute.bndlib.tests project. Says it is missing biz.aQute.bnd.test which is clearly there, as is the test folder.

image

**Pull request merged**